packages/core/src/lib/metadata/index.ts

```diff
@@ -132,7 +133,9 @@ class Metadata extends BaseProtocol implements IMetadata {
 }
 
 export function wakuMetadata(
-  shardInfo: ShardInfo
+  shardInfo: ShardInfo,
+  pubsubTopics: PubsubTopic[]
```

Comment:
why are we inputting both pubsubtopics and shardInfo? they can be converted both ways, right?

Reply:
They can but I'd like to keep conversion out of the core classes.

Reply:
Should we remove passing shardInfo then?

Reply:
Metadata needs to store `ShardInfo` because it encodes it into requests/responses for metadata.

Reply:
ah! does it feel redundant to you? happy to let it be for now if you're convinced to unblock the PR merge but i would aim to only provide either of the two

Reply:
I reviewed how it's used and realized it's only ever called if sharding is enabled, so just `ShardInfo` is sufficient. Updated in latest commit